api/channel.go

```diff
@@ -965,7 +866,27 @@ func getChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 			w.Write([]byte(data.ToJson()))
 		}
 	}
+}
 
+func SetActiveChannel(userId string, channelId string) *model.AppError {
+	status, err := GetStatus(userId)
+	if err != nil {
+		status = &model.Status{userId, model.STATUS_ONLINE, false, model.GetMillis(), channelId}
+	} else {
+		status.ActiveChannel = channelId
+		if !status.Manual {
+			status.Status = model.STATUS_ONLINE
+		}
+		status.LastActivityAt = model.GetMillis()
+	}
+
+	AddStatusCache(status)
+
+	if result := <-Srv.Store.Status().SaveOrUpdate(status); result.Err != nil {
```

Comment:
This isn't new code I just moved it here. That said, I think the DB write can be completely removed in this instance now that the changes in statuses are broadcast across HA. If the status is automatic or manual shouldn't matter since that should have been taken care of by the normal method of statuses being set.